File: gestisoft-server/src/main/java/py/edu/gestisoft/model/base/Vendedor.java
package py.edu.gestisoft.model.base;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.PrePersist;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.Setter;

@Entity
@Table(name = "VENDEDOR")
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Vendedor {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ID_VENDEDOR")
	private Long id;

	@NonNull
	@Column(name = "ESTADO")
	private Boolean estado;

	@NonNull
	@Column(name = "NOMBRE")
	private String nombre;

	@Column(name = "CI")
	private String ci;

	@Column(name = "CELULAR")
	private String celular;

	@Column(name = "LINEA_BAJA")
	private String lineaBaja;

	@Column(name = "CIUDAD")
	private String ciudad;

	@Column(name = "DIRECCION")
	private String direccion;

	@Column(name = "OBSERVACION")
	private String observacion;

	@Column(name = "FECHA_ALTA")
	private LocalDateTime fechaAlta;

	@PrePersist
	void setFechaAlta() {
		this.fechaAlta = LocalDateTime.now();
	}
}

File: gestisoft_windows/lib/app/modules/vendedor/models/vendedor.dart
import 'package:mobx/mobx.dart';
part 'vendedor.g.dart';

class Vendedor = VendedorBase with _$Vendedor;

abstract class VendedorBase with Store {
  int? id;
  bool? estado;
  String? nombre;
  String? ci;
  String? celular;
  String? lineaBaja;
  String? ciudad;
  String? direccion;
  String? observacion;
  String? fechaAlta;

  VendedorBase(
      {id,
      estado,
      nombre,
      ci,
      celular,
      lineaBaja,
      ciudad,
      direccion,
      observacion,
      fechaAlta});

  VendedorBase.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    estado = json['estado'];
    nombre = json['nombre'];
    ci = json['ci'];
    celular = json['celular'];
    lineaBaja = json['lineaBaja'];
    ciudad = json['ciudad'];
    direccion = json['direccion'];
    observacion = json['observacion'];
    fechaAlta = json['fechaAlta'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['estado'] = estado;
    data['nombre'] = nombre;
    data['ci'] = ci;
    data['celular'] = celular;
    data['lineaBaja'] = lineaBaja;
    data['ciudad'] = ciudad;
    data['direccion'] = direccion;
    data['observacion'] = observacion;
    data['fechaAlta'] = fechaAlta;
    return data;
  }

  VendedorBase.nuevo({this.estado = true, this.nombre = ""});
}
